src/serval.py

```diff
@@ -1544,25 +1622,73 @@ def serval():
                #pause()
                if it < n_iter: ind[ind] *=  okmap
 
-            if ofacauto:
-               # BIC to get optimal knot spacing (smoothing)
-               chired = []
-               BIC = []
-               K = np.logspace(np.log10(10), np.log10(ntpix), dtype=int)
-               for Ki in K:
-                  smod, ymod = spl.ucbspl_fit(wmod[ind], mod[ind], we[ind], K=Ki, lam=pspllam, mu=mu, e_mu=e_mu, e_yk=True, retfit=True)
-                  chi = ((mod[ind] - ymod)**2*we[ind]).sum()
-                  chired += [ chi / (we[ind].size-Ki)]
-                  BIC += [ chi + np.log(we[ind].size)*Ki]
-
-               Ko = K[np.argmin(BIC)]
-               smod, ymod = spl.ucbspl_fit(wmod[ind], mod[ind], we[ind], K=Ko, lam=pspllam, mu=mu, e_mu=e_mu, e_yk=True, retfit=True)
-               print("K=%d " % Ko, end='')
-
-               if 0:
-                  gplot2(K, BIC, 'w lp,', Ko, min(BIC), 'lc 3 pt 7')
-                  gplot(wmod[ind], mod[ind], 'w d,', smod.osamp(10), 'w lp ps 0.3,', smod.xk, smod(), 'w p')
-                  #pause()
+               if ofacauto:
```

Comment:
The intend has increased meaning it is now executed in each `niter` clipping loop (https://github.com/hlruh/serval/blob/767f8f41197dc586cf72319ea33c2b082cc344f1/src/serval.py#L1552).
Please revert.

Reply:
fixed